### ISSUE TITLE
Symbolic backend fails on rotates of zero-width bitvectors

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#397]
Here's the example:
```
Cryptol> 0 <<< 3 == []
True
Cryptol> :prove 0 <<< 3 == []
*** An error occurred.
***  Failed to complete the call to Z3
***  Executable   : "/Users/huffman/.bin/z3"
***  Options      : -nw -in -smt2
***  Exit code    : 1
***  Solver output: 
***  ==============================================================================
***  (error "line 12 column 29: bit-vector size must be greater than zero")
***  unsat
***  ==============================================================================
***  Giving up..

(simulation: 0.000s, export: 0.000s, Z3: 0.005s)
```
The symbolic backend should be able to reduce this to `True` by constant-folding. It should never generate symbolic bitvector variables of width 0 in the first place.

**Comment by brianhuffman** [comment]
Interestingly, this works fine if the rotate is replaced with a shift.

**Comment by brianhuffman** [comment]
It appears that this is due to a bug in the SBV.Dynamic library (`svRotateLeft` and `svRotateRight` in particular):

```
Prelude Data.SBV.Dynamic> svShiftLeft (svInteger (KBounded False 0) 0) (svInteger (KBounded False 3) 3)
0 :: SWord0
Prelude Data.SBV.Dynamic> svRotateLeft (svInteger (KBounded False 0) 0) (svInteger (KBounded False 3) 3)
<symbolic> :: SWord0
```

Note that the left shift is properly constant-folded, but the left rotate results in a symbolic value (even though both arguments are concrete).

**Comment by atomb** [comment]
This is fixed with the update to SBV 7.0.
